text/0000-doctest-name.md

```diff
@@ -0,0 +1,143 @@
+- Feature Name: doctest-name
+- Start Date: 2010-10-07
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Allow giving documentation tests names for identification in test runner
+output by including `name=NAME` in doctest code block info strings.
+
+# Motivation
+[motivation]: #motivation
+
+When doctests run, the test runner prints the name of the module that contains
+them and the line number that they start on. If a module contains many
+doctests, it is hard to identify which doctest is which in the test runner
+output by line number alone.
+
+By giving users the option to give names to doctests, identifying individual
+doctests in test runner output will be easier. Additionally, users could re-run
+specific tests using `cargo test --doc -- $TEST_NAME`.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Doctests are written as fenced markdown code blocks that start and end
+with lines containing three backticks.
+
+The first set of backticks may be followed by an info string. For example, to
+tell the Rust compiler to ignore the doctest:
+
+````rust
+
+// This doctest won't run.
+```ignore
+assert_eq!(2 + 2, 4);
+```
+
+````
+
+Normally, doctests do not have names, and when run, the test runner will
+identify them by their test binary, module name, and line number:
+
+```
+   Doc-tests foo
+
+running 1 test
+test src/lib.rs - bar (line 37) ... ok
+
+test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
+```
+
+If you'd like to have the test runner print a more informative name, perhaps
+because there are many doctests in a single file, you can put `name=NAME`,
+where `NAME` is the name the test should have, in the doctest's info string.
```

Comment:
```suggestion
because there are many doctests in a single file, you can put `name=IDENT`,
where `IDENT` is the name the test should have, in the doctest's info string.
```

And likewise elsewhere in the document: let's explicitly specify that the name must be an identifier. This also disambiguates cases like `name=ident,should_panic`.

Reply:
Good call, done!